### DISCUSSION TITLE
zigbee2mqtt without zigbee. Possible?

### DISCUSSION THREAD

**Discussion by bartplessers** [Koenkk/zigbee2mqtt#7400] [General]
Hi,

Can I run Z2M without zigbee-stick? I have several instances of hass running on different rpi’s. Only one of them can run Z2M (the one with the zigbee stick attached). I want to run the frontend of zigbee2mqtt on the others.

If I could run Z2M with frontend, but without any stick attached, that would be great.

see also
https://github.com/nurikk/z2m-frontend/issues/506

**Comment by Koenkk** [comment]
No not possible.

**Reply by bartplessers** [reply]
Any timeframe when it will? 🤣

Just kidding. But it has a use case: my whole domotica is relying an a rpi with a sd-card of 10 eur. Does not feel comfortable, so I have a "hot spare" with same config, even with zigbee2mqtt installed and configured. Just in case my other rpi crashes. 
It would be nice to have z2m frontend running there to. 

Other use case: I have splitted my setup: 
- backend = rpi + mqtt + z2m
- frontend = rpi + hass
It would be nice to have the z2m frontend available there. It's much better than the z2massistant frontend that I have to run now

Do I have to continue... ?

Grtz
B

**Reply by DubhAd** [reply]
> Just in case my other rpi crashes.
> It would be nice to have z2m frontend running there to.

But, if the back end crashes, what does it matter if you've got a useless UI? You can't do anything with it.

The solutions to your concern are:
1. Don't use an SD card
2. Use a long (write) life SD card, like an industrial one
3. Use tools like `rpi-clone` to take a live image of your system
4. Don't use a Pi 

All of this is in addition to your normal backups, obviously.

**Reply by bartplessers** [reply]
True, but my use case still remains. 
If my backend crashes, I just can switch my stick to the frontend wich will become front- and backend than. Just change my DNS and all MQTT-messages will go to the new broker. Z2M config is already duplicated, so a failover is done very easy: switch stick and DNS, done.

I'm not a programmer, but I think an implementation of what I asked does not mean much work, and really has some benefits for at least some of us. 

Grtz
B

**Comment by araczkowski** [comment]
IMHO, this can be nice improvement for zigbee2mqtt

it will be good to have possibility to do the configuration from web app (without yaml), this is the way to be able to reach less advanced users - the connection with adapter can be configured in web app:

![image](https://user-images.githubusercontent.com/1263968/118696602-bde6bb00-b80e-11eb-90ce-1a30f86b666b.png)

for example Home Assistant is moving to app from yaml and this is very good move

I don't know how this is technically done, but it is possible to start zwavejs2mqtt (node app) without connection to zwave adapter and configure the connection from app 

![image](https://user-images.githubusercontent.com/1263968/118697006-2766c980-b80f-11eb-95ff-8810490f54a3.png)

**Comment by araczkowski** [comment]
this is how the ziwavejs2mqtt is starting for the first time, without adapter

![image](https://user-images.githubusercontent.com/1263968/118942542-1cb54d00-b953-11eb-92b1-26ebee0f1ab0.png)

all the configuration can be done in web app 

![image](https://user-images.githubusercontent.com/1263968/118942683-3fdffc80-b953-11eb-8e0f-62a550c2c1de.png)


I think zwavejs2mqtt is a copycat of zigbee2mqtt
but have to say, thay do this one thing (start app without adapter) :)

**Reply by timoline** [reply]
I think this is a really nice idea !
I already proposed a different start sequense earlier, which could help for this 
https://github.com/Koenkk/zigbee2mqtt/discussions/5653

**Comment by asystentka-jolka** [comment]
I know that the fronted was added not so long ago, and that it is optional. But start the frontend first, than the zigbee and mqtt... is brilliant idea :)

The frontend should start first then the mqtt and zigbee connection. This way we can configure the all the settings from web app.

Is it technically possible?

**Reply by Koenkk** [reply]
Yes that is possible, I will consider this.

**Reply by bartplessers** [reply]
@Koenkk that is super!
please consider also a parameter / check that z2m does not start if no stick attached.

regards,
B

**Reply by Koenkk** [reply]
@bartplessers yes, so the idea is that z2m can start without adapter and MQTT connection. Then the configuration can be done from the frontend.

**Comment by asystentka-jolka** [comment]
@Koenkk I noticed that this works diferent for deconz adapter then for zstack

## deconz
The app is still unable to start without adapter, but after the app is started I can disconnect the adapter and the app is still working.  

1. start the app
2. remove the adapter from USB
![image](https://user-images.githubusercontent.com/43966761/125460458-a2f2d198-36b9-4139-98d0-9cc8762fe0ee.png)

3. plug the USB adapter again
![image](https://user-images.githubusercontent.com/43966761/125460298-a034da24-b6be-48f6-a176-a12da4bf3663.png)

works without need to restart the service :) 
In my opinion this is better then for zstack

## zstack
after the adapter is removed from USB the node app is stopped

**Reply by Koenkk** [reply]
In my opinion this is undesired behaviour since the user has no clue wether z2m is still working properly.

**Comment by kladderadeng** [comment]
I was not able to find a / the parameter in the docs that allows startup without an adapter connected. I suppose this did not get implemented? I just wanted to configure a new instance "off-site", before rolling it out. So having the Web interface for that even with not adapter connected / discoverable would have been neat :-)

**Comment by phyber** [comment]
Found this issue just now while I'm migrating Zigbee2MQTT from Docker to Kubernetes. Being able to view the frontend and confirm configuration without having the USB adapter connected would be useful. (I know I can view the ConfigMaps, etc, but being able to view the frontend would be a nice final check).
